algorithmic_efficiency/workloads/imagenet/imagenet_jax/input_pipeline.py

```diff
@@ -156,16 +158,18 @@ def preprocess_for_train(image_bytes,
   Returns:
     A preprocessed image `Tensor`.
   """
-  crop_rng, flip_rng = tf.random.experimental.stateless_split(rng, 2)
+  # Note (runame): Cannot be done in graph mode, i.e. during ds.map().
+  # Alternative?
+  # crop_rng, flip_rng = tf.random.experimental.stateless_split(rng, 2)
```

Comment:
@znado This doesn't work in graph mode (see my comment), what do you think is the best alternative?

Reply:
Something like "stateless_split iterating over `tf.Tensor` is not allowed in Graph execution" (according to my google search history lol). It's especially weird because [here](https://github.com/mlcommons/algorithmic-efficiency/pull/81/files/c4a07ac76aa311af883886579344581cc4f9394c#diff-60eed9f70147177b1d33b389535916e760ef922b378efac41e537ef089b85c95R225), [`stateless_fold_in()`](https://github.com/tensorflow/tensorflow/blob/v2.9.1/tensorflow/python/ops/stateless_random_ops.py#L204-L242) works just fine and is also called inside of the `ds.map()` call, i.e. is also executed in graph mode. And it is calling the same underlying function. So the issue might be related to the shape argument in `stateless_random_uniform()`, but I haven't looked into it further.